src/App.tsx

```diff
@@ -1,73 +1,148 @@
-import React from 'react';
+import React, {
+  useCallback,
+  useMemo,
+  useState,
+  useRef,
+  useEffect,
+  MutableRefObject,
+} from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
+import { Person } from './types/Person';
+
+const delay = 300;
```

Comment:
The debounce delay is currently hardcoded as a constant. To meet the task requirements, it should be customizable via props.